src/const.js

```diff
@@ -162,7 +162,7 @@ export const MESSAGES_JSON = 'messages.json';
 export const LOCALES_DIRECTORY = '_locales';
 
 // This is a string, since it has to be matched globally on a message string.
-export const MESSAGE_PLACEHOLDER_REGEXP = '\\$([a-z0-9_@]+)\\$';
+export const MESSAGE_PLACEHOLDER_REGEXP = '\\$([a-z0-9_@\\.]+)\\$';
```

Comment:
I don't think that we should change this regular expression (as well as the one in  src/schema/messages.json at line 30), because the placeholder are not validated by Firefox when the extension is installed (and so it will not raise an error), but a similar regular expression is used to apply the placeholder here:

https://searchfox.org/mozilla-central/rev/3abf6fa7e2a6d9a7bfb88796141b0f012e68c2db/toolkit/components/extensions/ExtensionCommon.jsm#1711

and so the placeholders that don't match that regexp will be just ignored.

Nevertheless, it seems to me that we should add `A-Z` to the ranges of characters matched by this regular expression (the placeholder matches besides its "case", then the matched placeholder is normalized to lowercase and retrieved from the map of the placeholders, also is also normalized to lowercase).

Reply:
> and so the placeholders that don't match that regexp will be just ignored.

Interesting, I'll test this given that we have user reports that these messages "simply work in Firefox"

> Nevertheless, it seems to me that we should add A-Z to the ranges of characters matched by this regular expression (the placeholder matches besides its "case", then the matched placeholder is normalized to lowercase and retrieved from the map of the placeholders, also is also normalized to lowercase).

That's a good call!